src/BitStringAddresses.jl

```diff
@@ -784,6 +785,116 @@ function check_consistency(b::BoseFS{N,M,A}) where {N,M,A<:Union{BSAdd64,BSAdd12
   leading_zeros(b.bs.add) ≥ numBits(A) - numBits(b) ||  error("Ghost bits detected in $b.")
 end
 
+
+
+
+#################################
+"""
+    BoseFS2C{NA,NB,M,AA,AB} <: BosonicFockStateAddress <: BitStringAddressType
+
+Address type that constructed with two [`BoseFS{N,M,A}`](@ref). It represents a
+Fock state with two components, e.g. two different species of bosons with particle
+number `NA` from specie A and particle number `NB` from specie B. The number of
+orbitals `M` is ecpacted to be the same for both components.
+"""
+struct BoseFS2C{NA,NB,M,AA,AB} <: BosonicFockStateAddress
```

Comment:
We should probably not subtype this to `BosonicFockStateAddress`:

```
help?> BitStringAddresses.BosonicFockStateAddress
  BosonicFockStateAddress <: BitStringAddressType

  Supertype representing a bosonic Fock state. Implement numModes() and numParticles().
```
Rather, subtype to `BitStringAddressType`.

Reply:
It would be nice to have a constructor that would allow you to write
`add = BoseFS2C((1,0,3),(2,1,0))`.

Also, it would nice if then `add` would print in the same way, such that the output can be copied and pasted for assignments (see how this was done adding `Base.show()` methods).

These things are optional, but would be nice in making your scripts (and interactive work) more readable.